html/browsers/history/the-location-interface/location-prototype-setting-same-origin-domain.sub.html

```diff
@@ -0,0 +1,59 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>[[SetPrototypeOf]] on a Location object should not allow changing its value: cross-origin, but same-origin-domain, Location</title>
+<link rel="help" href="http://html.spec.whatwg.org/multipage/#location-setprototypeof">
+<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<iframe src="//{{domains[www]}}:{{ports[http][1]}}/common/domain-setter.sub.html"></iframe>
```

Comment:
Hmm, I'm not sure what that would look like in practice. Although I do agree there's a decent amount of duplicated code, I'm also trying to avoid too much logic in the tests... I'm happy to change it up though if you can describe in a bit more detail.